lightning/src/ln/chan_utils.rs

```diff
@@ -318,7 +318,8 @@ impl_writeable!(ChannelPublicKeys, 33*5, {
 
 
 impl TxCreationKeys {
-	pub(crate) fn new<T: secp256k1::Signing + secp256k1::Verification>(secp_ctx: &Secp256k1<T>, per_commitment_point: &PublicKey, a_delayed_payment_base: &PublicKey, a_htlc_base: &PublicKey, b_revocation_base: &PublicKey, b_htlc_base: &PublicKey) -> Result<TxCreationKeys, secp256k1::Error> {
+	/// Create a new TxCreationKeys
+	pub fn new<T: secp256k1::Signing + secp256k1::Verification>(secp_ctx: &Secp256k1<T>, per_commitment_point: &PublicKey, a_delayed_payment_base: &PublicKey, a_htlc_base: &PublicKey, b_revocation_base: &PublicKey, b_htlc_base: &PublicKey) -> Result<TxCreationKeys, secp256k1::Error> {
```

Comment:
Hmm, interesting, right. We don't want to expose our own internally-derived `TxCreationKeys`'s keys, but its totally find if an external signer uses this code to derive their own copy. I'm not sure how to capture that appropriately except to use a whole different struct :/

Reply:
`TxCreationKeys` are used in the `ChannelKeys` interface, so we both need to read the struct when we get them passed in, and to construct the struct when we use `InMemoryChannelKeys` internally.

Reply:
(`TxCreationKeys` are passed directly in `sign_remote_commitment` and indirectly via `LocalCommitmentTransaction` to `sign_local_*`)

Reply:
ah, only the `per_commitment_point` really needs to be public, and it already is.  the rest can be recomputed.

Reply:
I've eliminated the commit that makes the fields public and updated our side to remove that use.  It was pretty straightforward.

Reply:
Hmm, right, but you actually pointed out a broader issue that we screwed up the API. I think we may still need to add a newtype to avoid users being able to indirectly access the innards of TxCreationKeys from the cached values without it being explicit to them.

eg via `get_htlc_redeemscript`, a bunch of the methods in `LocalCommitmentTransaction`, etc.